### PR TITLE
Do not stop watching on general overflow event

### PR DESCRIPTION
**Pull request by lptr**
And also demote message about overflows to `INFO` level.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QF please.

File: subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java [hunk 1]
```diff
@@ -86,15 +87,20 @@ public void handleChangeEvent(FileWatchEvent.ChangeType type, String absolutePat
 
                             @Override
                             public void handleUnknownEvent(String absolutePath) {
+                                LOGGER.error("Received unknown event for {}", absolutePath);
```
Comment (wolfs) [inline subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java:90]: `LOGGER.WARN` should be good enough here. I don't know if it really matters, though.

File: subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java [hunk 2]
```diff
@@ -86,15 +87,20 @@ public void handleChangeEvent(FileWatchEvent.ChangeType type, String absolutePat
...
                                 fileWatchingStatistics.updateAndGet(MutableFileWatchingStatistics::unknownEventEncountered);
-                                handler.handleLostState();
+                                handler.stopWatchingAfterError();
                             }
 
                             @Override
                             public void handleOverflow(FileWatchEvent.OverflowType type, @Nullable String absolutePath) {
                                 if (absolutePath == null) {
-                                    handler.handleLostState();
+                                    LOGGER.info("Overflow detected, invalidating all watched hierarchies");
```
Comment (wolfs) [inline subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java:98]: Do we do some logging on the native side as well? Do we maybe need to change the log-level there?

**Review by wolfs — COMMENTED** [review]
I left some comments.

File: subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java (hunk 1 above)
Comment (lptr) [reply subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java:90]: We've used `.error()` for the actual error event. I think it's important to be consistent, and we are talking about an error here I think, though `warn()` is fine with me if you feel strongly about it.

Comment (wolfs) [reply subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java:90]: For me this doesn't seem like an error, though I also don't feel strongly about moving it to warn.

File: subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java (hunk 2 above)
Comment (lptr) [reply subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java:98]: We log at `INFO` on the native side:

https://github.com/gradle/native-platform/blob/24b02d4d6022d8ed6a6986c1ae9e1e77c4951d6f/file-events/src/file-events/cpp/generic_fsnotifier.cpp#L83-L89

But the messages cover different purposes, so I think it's okay to log both.

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**